apps/emqx_bridge_azure_blob_storage/src/emqx_bridge_azure_blob_storage_action_schema.erl

```diff
@@ -0,0 +1,310 @@
+        }
+    );
+fields(direct_parameters) ->
+    [
+        {mode, mk(direct, #{required => true, desc => ?DESC("direct_mode")})},
+        %% Container in the Azure Blob Storage domain, not aggregation.
+        {container,
+            mk(
+                emqx_schema:template_str(),
+                #{
+                    desc => ?DESC("direct_container_template"),
+                    required => true
+                }
+            )},
+        {blob,
+            mk(
+                emqx_schema:template_str(),
+                #{
+                    desc => ?DESC("direct_blob_template"),
+                    required => true
+                }
+            )},
+        {content,
+            mk(
+                emqx_schema:template(),
+                #{
+                    required => false,
+                    default => <<"${.}">>,
+                    desc => ?DESC("direct_content_template")
+                }
+            )}
+        | fields(common_action_parameters)
+    ];
+fields(aggreg_parameters) ->
+    [
+        {mode, mk(aggregated, #{required => true, desc => ?DESC("aggregated_mode")})},
+        {aggregation, mk(ref(aggregation), #{required => true, desc => ?DESC("aggregation")})},
+        %% Container in the Azure Blob Storage domain, not aggregation.
+        {container,
+            mk(
+                string(),
+                #{
+                    desc => ?DESC("aggregated_container_name"),
+                    required => true
+                }
+            )},
+        {blob,
+            mk(
+                emqx_schema:template_str(),
+                #{
+                    desc => ?DESC("aggregated_blob_template"),
+                    required => true
+                }
+            )}
+        | fields(common_action_parameters)
+    ];
+fields(aggregation) ->
+    [
+        emqx_connector_aggregator_schema:container(),
+        %% TODO: Needs bucketing? (e.g. messages falling in this 1h interval)
+        {time_interval,
+            hoconsc:mk(
+                emqx_schema:duration_s(),
+                #{
+                    required => false,
+                    default => <<"1h">>,
+                    desc => ?DESC("aggregation_interval")
+                }
+            )},
+        {max_records,
+            hoconsc:mk(
+                pos_integer(),
+                #{
+                    required => false,
+                    default => 1_000_000,
+                    desc => ?DESC("aggregation_max_records")
+                }
+            )},
+        {min_block_size,
+            mk(
+                emqx_schema:bytesize(),
+                #{
+                    default => <<"10mb">>,
+                    importance => ?IMPORTANCE_HIDDEN,
+                    required => true,
+                    validator => fun block_size_validator/1
+                }
+            )}
+    ];
+fields(common_action_parameters) ->
+    [
+        {max_block_size,
```

Comment:
Q: Is there anything this parameter may affect for direct uploads? The idea is to prematurely drop the event on the floor if it's for some reason >4GiB in size?

Reply:
Good point, currently we don't check it, but this limit indeed is Azure's maximum upload per block.
(More precisely, for a single "Put Block Blob" call, it's possible to send 5000 MiB, and each "Put Blob" is 4000 MiB)

Reply:
Added a check and test here: https://github.com/emqx/emqx/pull/13069/commits/a92eb3c60f89241460ae50dadfd0d65f85b8c1fa